### PR TITLE
Add the XPU device check for pipeline mode

### PR DESCRIPTION
**Pull request by yuanwu2017**
When setting xpu device for pipeline, It needs to use is_torch_xpu_available to load ipex and determine whether the device is available.

# What does this PR do?

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by ArthurZucker** [comment]
Hey! Feel free to ping me for a review when this is ready!

**Comment by yuanwu2017** [comment]
I tested the failed run_tests case in my local machine, but I got following warning. I think the case failure is not caused by my patch. Do you know how to run the case correctly? Thanks a lot.


(upstream) [yuanwu@skyocean transformers]$ pytest tests/test_modeling_flax_common.py
============================================================================================ test session starts =============================================================================================
platform linux -- Python 3.9.18, pytest-7.4.0, pluggy-1.0.0
rootdir: /mnt/disk4/yuanwu/workspace/transformers
configfile: pyproject.toml
plugins: anyio-3.5.0, xdist-3.5.0, timeout-2.2.0, hypothesis-6.92.2, dash-2.14.2
collected 0 items

============================================================================================== warnings summary ==============================================================================================
../../../../../home/yuanwu/.conda/envs/upstream/lib/python3.9/site-packages/_pytest/config/__init__.py:1373
  /home/yuanwu/.conda/envs/upstream/lib/python3.9/site-packages/_pytest/config/__init__.py:1373: PytestConfigWarning: Unknown config option: doctest_glob

    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================================= 1 warning in 0.38s =============================================================================================

**Comment by yuanwu2017** [comment]
@ArthurZucker I think the patch is ready. Please review. Thanks.

**Comment by ArthurZucker** [comment]
Yes sorry for the delay!

File: src/transformers/pipelines/base.py [hunk 1]
```diff
@@ -805,18 +810,37 @@ def __init__(
 
         if is_torch_available() and self.framework == "pt":
             if isinstance(device, torch.device):
+                if device.type == "xpu" and not is_torch_xpu_available(check_device=True):
+                    logger.warning("XPU is not available. Use CPU!")
```
Comment (ArthurZucker) [inline src/transformers/pipelines/base.py:814]: Not 100% sure we want to add a warning, raising an error might be better no?

**Review by ArthurZucker — APPROVED** [review]
LGTM! Thanks you for adding xpu checks

File: src/transformers/pipelines/base.py [hunk 2]
```diff
@@ -805,18 +810,37 @@ def __init__(
...
                 self.device = torch.device(device)
             elif device < 0:
                 self.device = torch.device("cpu")
-            else:
+            elif is_torch_cuda_available():
                 self.device = torch.device(f"cuda:{device}")
+            elif is_torch_xpu_available(check_device=True):
+                self.device = torch.device(f"xpu:{device}")
+            else:
+                self.device = torch.device("cpu")
```
Comment (yuanwu2017) [inline src/transformers/pipelines/base.py:829]: @ArthurZucker  If it is consistent, then it also needs to raise error here.

**Comment by yuanwu2017** [comment]
Hi @ArthurZucker  
Done for raising error when device is not available.

File: src/transformers/pipelines/base.py [hunk 3]
```diff
@@ -805,18 +810,35 @@ def __init__(
 
         if is_torch_available() and self.framework == "pt":
             if isinstance(device, torch.device):
+                if device.type == "xpu" and not is_torch_xpu_available(check_device=True):
+                    raise ValueError(f"{device} is not available")
```
Comment (ArthurZucker) [inline src/transformers/pipelines/base.py:814]:
```suggestion
                    raise ValueError(f"{device} is not available, you should use `device="cpu"` instead")
```

File: src/transformers/pipelines/base.py [hunk 4]
```diff
@@ -805,18 +810,35 @@ def __init__(
...
                 self.device = torch.device(device)
             elif device < 0:
                 self.device = torch.device("cpu")
-            else:
+            elif is_torch_cuda_available():
                 self.device = torch.device(f"cuda:{device}")
+            elif is_torch_xpu_available(check_device=True):
+                self.device = torch.device(f"xpu:{device}")
+            else:
+                raise ValueError(f"xpu:{device} and cuda:{device} are not available")
```
Comment (ArthurZucker) [inline src/transformers/pipelines/base.py:827]:
```suggestion
                raise ValueError(f"{device} unrecognized or not available.")
```

**Review by ArthurZucker — COMMENTED** [review]
Would go with something like this

**Comment by yuanwu2017** [comment]
Thanks @ArthurZucker 
Done.

**Comment by ArthurZucker** [comment]
Thanks, let's make sure the CI is green!

**Comment by amyeroberts** [comment]
@yuanwu2017 There was a fix merged into main which resolves the currently failing tests - could you rebase to include these and trigger an new CI run?

**Pull request merged**